Doc tabstop clarify (#17381)
vim/vim: runtime/doc/options.txt

				keysym 22 = BackSpace
<			You need to restart for this to take effect.

> 					 	*using-tabs* *tabs-usage*

Reviewer: Every time documentation tags are introduced or removed, the  
tags file should be regenerated:

```sh
cd runtime/doc
make tags
git add tags
```

Author: Thanks for letting me know. Done

There are five main ways to use tabs in Vim:

> 1. Always keep 'tabstop' at 8, set 'softtabstop' and 'shiftwidth' to 4

Reviewer 2: ```suggestion
1. Always keep 'tabstop' at 8, custome the 'softtabstop' and 'shiftwidth' value to e.g. 4
```

Author: I clarified some of the points of the section in the latest commit.

>    (or 3 or whatever you prefer) and use 'noexpandtab'.  Then Vim

Reviewer 2: ```suggestion
   and use 'noexpandtab'.  Then Vim
```

   will use a mix of tabs and spaces, but typing <Tab> and <BS> will

>    behave like a tab appears every 4 (or 3) characters.

Reviewer 2: ```suggestion
   behave like a tab appears every 4 characters.
```

   This is the recommended way, the file will look the same with other
   tools and when listing it in a terminal.

2. Set 'softtabstop' and 'shiftwidth' to whatever you prefer and use
   'expandtab'.  This way you will always insert spaces.  The
   formatting will never be messed up when 'tabstop' is changed (leave
   it at 8 just in case).  The file will be a bit larger.

>    You do need to check if no Tabs exist in the file.  You can get rid

Reviewer 2: Not sure what you mean with "you do need to check if no tabs exist in the file".
```suggestion
   Vim won't add a real Tab to the file, but won't touch any already existing tab present in the file.
```

Author: I kept the original wording of that whole section. So I cannot tell you what the author behind "you do need to check if no tabs exist in the file" had in mind.

My idea was: I am not going to touch the recommendation section because if I do, then I need to think about: is ts=8 sts=4 sw=4 really the best recommendation in 2025 ?
And : why is the recommended way, not the default ?

Reviewer 2: okay fair

Author: In case I might want to polish that section, what would be the best way to start a conversation about what should be recommended / the default etc?

Reviewer: > You do need to check if no Tabs exist in the file.

Raw tabs are significant for column-wise alignment in string  
constants, e.g.:

```vim
	echo printf("%s	%s	%s", "a", "aa", "...")
"......^...............^.......^
	echo printf("%-7s %-7s %s", "a", "aa", "...")
	echo printf("\t\t...")
"......^
" :source %
" :%retab
" :write
" :source %
```

One should definitely search and replace any such tabs with  
`\t` or `\x09` or whatever is supported _before_ `:%retab`ing  
and starting to use spaces only.

Reviewer 2: yes, one needs to be careful when using `:retab` as this may mess up formatting. Same is true, when using `expandtab` and changing the tabstop value as this won't adjust (or destroy) the existing formatting.

Author: @zzzyxwvut I have found that exact remark in the user manual section 30.5
This lead me to think that this is the place people should be referred to, instead of a section within the options.

>    of them by first setting 'expandtab' and using `%retab!`, making

Reviewer 2: ```suggestion
   of them by first setting 'expandtab' and using |:retab!|, making
```

   sure the value of 'tabstop' is set correctly.

3. Set 'tabstop' and 'shiftwidth' to whatever you prefer and use
   'expandtab'.  This way you will always insert spaces.  The
   formatting will never be messed up when 'tabstop' is changed.
   You do need to check if no Tabs exist in the file, just like in the
   item just above.

> 4. Set 'tabstop' and 'shiftwidth' to whatever you prefer and use a

Reviewer 2: ```suggestion
4. Set 'tabstop' and 'shiftwidth' to whatever you prefer via a 
```

   |modeline| to set these values when editing the file again.  Only

>    works when using Vim to edit the file, other tools assume a tabstop

Reviewer 2: That: "other tools assume a tabstop is worth 8 spaces". is true for all five cases. So probably rather add a sentence at the end after those customizations like:

> Keep in mind that other tools (like `git` or `cat`) will assume a tabstop is worth 8 spaces, even configured differently in Vim.

Author: Correct. I have clarified this in the latest commit.

				keysym 22 = BackSpace
<			You need to restart for this to take effect.

> 					 	*using-tabs* *tabs-usage*

Reviewer: ```suggestion
						*using-tabs* *tabs-usage*
```

Watch closely, there is a space character [between tabs](https://cirrus-ci.com/task/5185974327574528).

Author: Well done!

						*'tabstop'* *'ts'*
'tabstop' 'ts'		number	(default 8)
			local to buffer
	Number of columns between two (virtual) horizontal tab stops.
	The value must be at least 1 and at most 9999.

	In insert mode, this is one of the settings that influence
	what characters Vim writes in a file when <Tab> is used
	and what characters get deleted when <BS> is used.
	See also the 'expandtab', 'softtabstop' and 'smarttab' options.

	It is also used to visually represent the Horizontal Tabulation
	characters (ASCII 9, caret notation: ^I) already present in a file.

> 	Example: with `ts=8` the following two lines

Reviewer 2: ```suggestion
	Example: with `:set tabstop=8` the following two lines
```

'tabstop' 'ts'		number	(default 8)
			local to buffer
	Number of columns between two (virtual) horizontal tab stops.
	The value must be at least 1 and at most 9999.

	In insert mode, this is one of the settings that influence
	what characters Vim writes in a file when <Tab> is used
	and what characters get deleted when <BS> is used.
	See also the 'expandtab', 'softtabstop' and 'smarttab' options.

	It is also used to visually represent the Horizontal Tabulation
	characters (ASCII 9, caret notation: ^I) already present in a file.
	Example: with `ts=8` the following two lines
>
		1234^I9
		1234    9
<
	will display identically on screen. See also the 'list' option.

	This setting does not influence how Vim represents the Vertical
	Tabulation characters (ASCII 11) already present in a file; they will
	always be represented by their caret notation `^K`.

	If Vim is compiled with the |+vartabs| feature then the value of
	'tabstop' will be ignored if |'vartabstop'| is set to anything other
	than an empty string.

	Note: Setting 'tabstop' to any other value than 8 can make your file
	appear wrong in many places, e.g., when printing it.
	See also |using-tabs| for recommendations.

> 

Reviewer 3: Too many contiguous blank lines. 2 should be enough.

Author: Thanks. Fixed in latest commit.
